pkgs/top-level/nim-packages.nix

```diff
@@ -1,12 +1,14 @@
-{ lib, pkgs, stdenv, newScope, nim, fetchFromGitHub }:
+{ lib, pkgs, stdenv, newScope, nim, fetchFromGitHub, buildPackages }:
 
 lib.makeScope newScope (self:
   let callPackage = self.callPackage;
   in {
     inherit nim;
     nim_builder = callPackage ../development/nim-packages/nim_builder { };
     buildNimPackage =
-      callPackage ../development/nim-packages/build-nim-package { };
+      callPackage ../development/nim-packages/build-nim-package {
+        inherit (buildPackages.buildPackages.nimPackages) nim_builder;
```

Comment:
Always building for target makes sense, but `buildInputs` seems to grab packages from `buildPackages` but for `depsBuildBuild` I manually need to take packages from `buildPackages.buildPackages`?

Reply:
maybe you should use makeScopeWithSplicing

Reply:
What would be the difference?

Reply:
makeScope breaks splicing, ie the mechanism that makes `nativeBuildInputs = [ foo ]` and `buildInputs = [ foo ]` pick a different derivation.

Reply:
That sounds good, but `overrideScope'` then is missing?

Reply:
I don't think makeScopeWithSplicing is a drop-in replacement, but of course it's not documented...